pythonlib/src/mill/pythonlib/PythonModule.scala

```diff
@@ -25,52 +44,65 @@ trait PythonModule extends Module {
     Task.traverse(moduleDeps)(_.typeCheck)()
 
     os.call(
-      (pythonExe().path, "-m", "mypy", "--strict", sources().path),
+      (pythonExe().path, "-m", "mypy", "--strict", sources().map(_.path)),
       stdout = os.Inherit,
       cwd = T.workspace
     )
   }
 
-  def gatherScripts(upstream: Seq[(PathRef, PythonModule)]) = {
-    for ((sourcesFolder, mod) <- upstream) {
-      val destinationPath =
-        os.pwd / mod.millSourcePath.subRelativeTo(mill.api.WorkspaceRoot.workspaceRoot)
-      os.copy.over(sourcesFolder.path / os.up, destinationPath)
-    }
-  }
-
   def run(args: mill.define.Args) = Task.Command {
-    gatherScripts(Task.traverse(moduleDeps)(_.sources)().zip(moduleDeps))
-
     os.call(
-      (pythonExe().path, sources().path / mainFileName(), args.value),
-      env = Map("PYTHONPATH" -> Task.dest.toString),
-      stdout = os.Inherit
+      (pythonExe().path, script().path, args.value),
+      env = Map(
+        "PYTHONPATH" -> transitiveSources().map(_.path).mkString(":"),
+        "PYTHONPYCACHEPREFIX" -> (T.dest / "cache").toString
```

Comment:
I'm doing this for now to avoid polluting source directories, and hence both `run` and `console` won't share any caches. In the future, it could be interesting to see if this can be used in a global cache somehow, so that we don't need to recreate it for every task.

Reply:
btw have this worked ???

when i was doing the same thing like giving paths to the files which are module dependencies but getting error like module doesn't exist

any correction ???

Reply:
https://github.com/com-lihaoyi/mill/tree/422ef4094600af1cc4c44027a8357dcd5e37bb01/example/extending/python/3-python-module-deps

Check this example tbh i haven't tested but i think this will not work coz when i was adding python support at that time i was getting module import error and then i tried to add `__init__.py` files it still didn't worked

if you have tested this already and it's working then i think this would be the best approach to handle this

Reply:
PS: i was copying the same structure like copying the files from on root folder like `foo/bar/src/bar.py` and `foo/src/foo.py` and then maintain the same structure in the `Task.dest`
and then call them in the `qux.py` where i include modules as :
`from foo.src.foo import someFunction` and same for bar `from foo.bar.src.bar import someFunction`

Reply:
and this is what we call `Experience` 

Hat's off, TBH i haven't even thought this

this is the best approach GREAT!!!